axum-macros/src/debug_handler.rs

```diff
@@ -225,7 +315,7 @@ fn check_output_impls_into_response(item_fn: &ItemFn) -> TokenStream {
             FnArg::Receiver(_) => None,
             FnArg::Typed(pat_ty) => {
                 let pat = &pat_ty.pat;
-                let ty = &pat_ty.ty;
+                let ty = specializer.specialize_default(&pat_ty.ty);
```

Comment:
It is unclear to me whether we need to check every combination or just a single combination. I think returning `impl Trait` requires a single concrete type underneath, but not sure if it can be different depending on the generic parameters (in which case maybe we should generate each combination?)

Reply:
The output type can be different per generic type, and in an `async fn`, it's practically *guaranteed to be* different since `async fn` always immediately create a future and return it, capturing all arguments (except maybe ones that are unused).